File: server/src/main/java/com/springboot/fstore/entity/Food.java
package com.springboot.fstore.entity;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor

@Entity
@Table(name = "foods")
public class Food extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String name;
    private String description;
    private String image;
    private Double price;
    private int quantity;
    private String category;
    private Boolean isDeleted;


    @OneToMany(mappedBy = "food", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<FoodSize> foodSizes;

}

File: models/Food.ts
export type FoodSize = {
  id: number;
  name: string;
  price: number;
  weight: number;
  note: string;
};

export type FoodCategory = {
  id: number;
  name: string;
  image: string;
};

export type Food = {
  id: number;
  name: string;
  description: string;
  images: string[];
  isDeleted: boolean;
  foodSizes: FoodSize[];
  category: FoodCategory;
  rating: number;
  tags: string[];
  createdDate: string;
};
